### DISCUSSION TITLE
Unhandled Exception

### DISCUSSION THREAD

**Discussion by ExpertInvest** [doghappy/socket.io-client-csharp#220] [Q&A]
Hi,
I trying to handle exception in my C# program if the server side has been closed, but the exception cause my C# application to crash.
This is the exception message I get:

> System.Threading.Tasks.TaskCanceledException: 'A task was canceled.'
> This exception was originally thrown at this call stack:
    System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
    System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
    SocketIOClient.Transport.TransportRouter.ConnectAsync()
    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
    SocketIOClient.SocketIO.ConnectAsync()
    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    SocketIOClient.SocketIO.ConnectAsync()
    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
    ...
    [Call Stack Truncated]

This is my code:

        class Program
        {
            static void Main(string[] args)
            {
                try
                {
                    Logger.LoggerLevel = LogLevel.Info;

                    ClientSocket clientSocket = new ClientSocket();
                    clientSocket.ConnectAsync();
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception: " + ex.Message);
                }
                Console.ReadKey();

            }
        }

        public class ClientSocket
        {
            public async void ConnectAsync()
            {
                try
                {
                    _client = new SocketIO("https://myserversideurl.com:7102/");
                    _client.OnConnected += Client_OnConnected;
                    _client.OnDisconnected += Client_OnDisconnected;
                    _client.OnError += Client_OnError;
 
                    _client.OnReconnected += Client_OnReconnected;
                    _client.OnReconnectAttempt += Client_OnReconnectAttempt;
                    _client.OnReconnectError += Client_OnReconnectError;
                    _client.OnReconnectFailed += Client_OnReconnectFailed;
                    _client.OnPing += Client_OnPing;
                    _client.OnPong += Client_OnPong;
                    _client.OnAny(AnyMessageHandler);
 
                    await _client.ConnectAsync();
 
                }
                catch (Exception ex)
                {
                     Console.WriteLine("Exception: " + ex.Message);
                }
             }
        }

How to produce:
1) I'm starting my server
2) Start the client
3) wait for connection
4) Close my server after 10 seconds
5) Getting Disconnection on the client side
6) App on client side (C#) crash after about 10 seconds because of unhandled exception.

Tried more things to find the exception, but nothing worked.

Thanks.

**Comment by doghappy** [comment]
I did not reproduce. Are you using the latest version? Can you provide a reproducible github repo

**Reply by ExpertInvest** [reply]
Hi,
I was using v3.0.1
I'll update to the latest version and will try to provide repo soon.

Thanks
